docs/porting/connectivity/Layer3IP.md

```diff
@@ -0,0 +1,144 @@
+
+User app code can create IP Layer3 in way shown below
+
+   	NetworkInterface *l3interface;
+   	l3interface =new L3IPInterface(L3IP::get_default_instance(), OnboardNetworkStack::get_default_instance());
+   	l3interface->connect();
+
+This attaches the default network stack (usually LWIP - the other alternative is Nanostack) to the specified L3IP driver, and provides all the `NetworkInterface` and `NetworkStack` APIs.
+
+Example of target device driver class that needs to be implemented.
+
+
+	class Cellular_driver_L3IP : public L3IP {... }
+
+	Cellular_driver_L3IP &Cellular_driver_L3IP::get_instance()
+	{
+    	static Cellular_driver_L3IP l3ip_test_driver;
+    	return l3ip_test_driver;
+	}
+
+	// Weak so a module can override
+	MBED_WEAK L3IP &L3IP::get_default_instance()
+	{
+    	return Cellular_driver_L3IP::get_instance();
+	}
+
+Following driver class example can be found in mbed-os/TESTS/network/l3ip directory.
+
+
+### Being the default Interface
+Currently EMAC based  interfaces are created by automated factories and can be default for applications.
+
+	NetworkInterface::get_default_instance()
+
+Can create ethernet, wifi, mesh or cellular depending on JSON config setting:
+
+    NETWORK_DEFAULT_INTERFACE_TYPE
+
+Interface created that way is set default. 
+
+IP Layer3  Interface is designed to  not use `NetworkInterface::get_default_instance` so it is set as non default on create.
+
+To set IP Layer3  as default interface new member of  `set_as_default()` can be used. This metod is not limited to IP Layer3 and can set any network interface as default.
+
+
+### Cellular interfaces
+
+As a Cellular interface, a little more work is necessary - at a minimum, you need to implement the extra configuration calls in `L3IPInterface`. This is because the network stacks and IP Layer3 APIs only relate to the basic data path - they have no knowledge of any other configuration mechanisms and assume they are already set up.
+
+To do this, create a C++ class that inherits from both `L3IPInterface` and base class for new IP Layer3 cellular interface solution. The `L3IPInterface` is a helper class that implements all the core `NetworkInterface` functionality for you. You then need to implement the extra configuration methods.
+
+
+You don't usually directly expose the `L3IP` driver class of a cellular  driver - it is not usually declared as `L3IP::get_default_instance`, but you would pass it to the constructor of your base `L3IPInterface`. This then makes it visible using the `getl3ip` method. 
+
+
+### OnboardNetworkStack
+
+The precise details of the `OnboardNetworkStack` API should not concern an L3IP driver writer - it provides the mechanism to bind a driver to a stack, and the APIs needed to implement a `NetworkInterface`, but this is handled  by `L3IPInterface`, either as a base class of your own `XXXInterface`.
```

Comment:
I don't understand this sentence:
> but this is handled  by `L3IPInterface`, either as a base class of your own `XXXInterface`.

Either ... or...

Is there an option?

Reply:
Corrected.